lib/ClusterShell/RangeSet.py

```diff
@@ -225,7 +253,8 @@ def dim(self):
 
     def _sorted(self):
         """Get sorted list from inner set."""
-        return sorted(set.__iter__(self))
+        # for mixed padding support, sort by both string length and index
+        return sorted(set.__iter__(self), key=lambda x: (len(x), int(x)))
```

Comment:
Just wondering if people would be more interesting in simple ordering (`int(x)`) like  `1`, `02`, `2`, `03` ... except if that is needed for folding algorithm?